tests/test_matrix.py

```diff
@@ -14,6 +14,64 @@ def test_strMatrix():
     assert mat.__str__() == "[{1.0}+{x},{2.0};{3.0},{4.0}]"
 
 
+def test_transposeMat():
+
+    mat = getTokens("[1, 2; \
+                    3,  4]")
+    matTranspose = mat.transposeMat()
+    assert matTranspose.__str__() == "[{1.0},{3.0};{2.0},{4.0}]"
+
+    mat = getTokens("[5,8,2;\
+                      12,30,9;\
+                      4,17,7]")
+    matTranspose = mat.transposeMat()
+    assert matTranspose.__str__() == "[{5.0},{12.0},{4.0};{8.0},{30.0},{17.0};{2.0},{9.0},{7.0}]"
+
+
+def test_isSquare():
+
+    mat = getTokens("[1, 2; \
+                      x, z]")
+    assert mat.isSquare()
+
+    mat = getTokens("[1, 2; \
+                      1, 3; \
+                      1, 4]")
+    assert not mat.isSquare()
+
+
+def test_isDiagonal():
```

Comment:
If possible also add a test case of initialising a diagonal matrix and check if its a diagonal matrix. It could be something like
```python
mat = DiagMat([3, 3], [1, 5, 2])
assert mat.isDiagonal()
```
Change the above test case accordingly if using token list list instead.
Do the same for identity matrix.